examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -384,18 +384,40 @@ export class TaskList extends DataObject implements ITaskList {
 	 * DataObject, by registering an event listener for changes to the task list.
 	 */
 	protected async hasInitialized(): Promise<void> {
-		const saved = this.root.get<IFluidHandle<SharedMap>>("savedData");
+		const [saved, draft] = await Promise.all([
+			this.root.get<IFluidHandle<SharedMap>>("savedData"),
+			this.root.get<IFluidHandle<SharedMap>>("draftData"),
+		]);
 		if (saved === undefined) {
 			throw new Error("savedData was not initialized");
 		}
 		this._savedData = await saved.get();
 
-		const draft = this.root.get<IFluidHandle<SharedMap>>("draftData");
 		if (draft === undefined) {
 			throw new Error("draftData was not initialized");
 		}
 		this._draftData = await draft.get();
 
+		// Check for other connected clients each time a new runtime is initialized
+		const connected = this.runtime.getAudience();
+		let clientSize = 0;
+		let loneClient: boolean = true;
+
+		// Increase the count of all non-summarizer clients
+		for (const [_, value] of connected.getMembers()) {
+			if (!value.scopes.includes("summary:write")) {
```

Comment:
Is it better to explicitly check if the client is interactive? 

Also, should we additionally check if the client running this check itself has interactive permissions?

Reply:
Rohan is right, a more explicit check would be valuable. I think `value.details.type` here should contain `"interactive"/"noninteractive"` (noninteractive meaning summarizer) for each client.

The arguably "correct" check for the interactive/noninteractive type of the _current_ client is probably possible but in a hacky way, so I think I'd avoid it and instead get that information through the audience. `connected.getMember(this.runtime.clientId)` should give us the entry for the current client and you can check the same property mentioned above to see if we are interactive or summarizer. And we should definitely do it, because we don't want the summarizer client to ever cause any changes; in a sense it's a read-only container.